android-appcoins-billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/helpers/AndroidBillingMapper.java

```diff
@@ -35,56 +32,64 @@ public PurchasesResult mapPurchases(Bundle bundle, String skuType) {
       String signature = signatureList.get(i);
       String id = idsList.get(i);
 
-      JsonObject jsonElement = jsonParser.parse(purchaseData)
-          .getAsJsonObject();
-      String orderId = jsonElement.get("orderId")
-          .getAsString();
-      String packageName = jsonElement.get("packageName")
-          .getAsString();
-      String sku = jsonElement.get("productId")
-          .getAsString();
-      long purchaseTime = jsonElement.get("purchaseTime")
-          .getAsLong();
-      int purchaseState = jsonElement.get("purchaseState")
-          .getAsInt();
-
-      String developerPayload = null;
-      if (jsonElement.get("developerPayload") != null) {
-        developerPayload = jsonElement.get("developerPayload")
-            .getAsString();
-      }
-      String token = null;
-      if (jsonElement.get("token") != null) {
-        token = jsonElement.get("token")
-            .getAsString();
-      }
-
-      if (token == null) {
-        token = jsonElement.get("purchaseToken")
-            .getAsString();
-      }
-      boolean isAutoRenewing = false;
-      if (jsonElement.get("autoRenewing") != null) {
-        isAutoRenewing = jsonElement.get("autoRenewing")
-            .getAsBoolean();
+      JSONObject jsonElement = null;
+      try {
+        jsonElement = new JSONObject(purchaseData);
+        String orderId = jsonElement.getString("orderId");
+        String packageName = jsonElement.getString("packageName");
+        String sku = jsonElement.getString("productId");
+        long purchaseTime = jsonElement.getLong("purchaseTime");
+        int purchaseState = jsonElement.getInt("purchaseState");
+
+        String developerPayload = null;
+        try {
+          if (jsonElement.getString("developerPayload") != null) {
+            developerPayload = jsonElement.getString("developerPayload");
+          }
+        } catch (org.json.JSONException e) {
+
```

Comment:
Avoid empty catch statements at all costs! At least always log the exception with an error. If it's not your code, check with the last person to have committed there what an exception there might mean and treat it accordingly.